test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIncrementalismTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class BuildIncrementalismTest : MSBuildIntegrationTestBase, IClassFixture<BuildServerTestFixture>
+    {
+        private readonly string _pipeName;
+
+        public BuildIncrementalismTest(BuildServerTestFixture fixture)
+        {
+            _pipeName = fixture.PipeName;
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task BuildIncremental_SimpleMvc_PersistsTargetInputFile()
+        {
+            // Arrange
+            var thumbprintLookup = new Dictionary<string, FileThumbPrint>();
+
+            // Act 1
+            var result = await DotnetMSBuild(
+                "Build",
+                $"/p:RazorCompileOnBuild=true /p:UseRazorBuildServer=true /p:_RazorBuildServerPipeName={_pipeName} /p:_RazorForceBuildServer=true");
+
+            var directoryPath = Path.Combine(result.Project.DirectoryPath, IntermediateOutputPath);
+            var filesToIgnore = new[]
+            {
+                // These files are generated on every build.
+                Path.Combine(directoryPath, "SimpleMvc.csproj.CopyComplete"),
+                Path.Combine(directoryPath, "SimpleMvc.csproj.FileListAbsolute.txt"),
+            };
+            var files = Directory.EnumerateFiles(directoryPath).Where(p => !filesToIgnore.Contains(p));
+            foreach (var file in files)
+            {
+                var thumbprint = GetThumbPrint(file);
+                thumbprintLookup[file] = thumbprint;
+            }
+
+            // Assert 1
+            Assert.BuildPassed(result);
+
+            // Act 2
```

Comment:
Consider putting this in a loop. Avoids the trouble of changing things if we fix up stuff in the future.  Also would be useful to add a comment that says why we're testing this i.e. we want to make sure all files on disk remain identical across multiple incremental builds.